homeassistant/components/yale_smart_alarm/__init__.py

```diff
@@ -21,6 +24,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         COORDINATOR: coordinator,
     }
 
+    # Migrate lock default code from config entry to lock entity
+    if config_entry_default_code := entry.options.get(CONF_CODE):
+        entity_reg = er.async_get(hass)
+        entries = er.async_entries_for_config_entry(entity_reg, entry.entry_id)
+        for entity in entries:
+            if entity.entity_id.startswith("lock"):
+                entity_reg.async_update_entity_options(
+                    entity.entity_id,
+                    LOCK_DOMAIN,
+                    {CONF_DEFAULT_CODE: config_entry_default_code},
+                )
+        new_options = entry.options.copy()
+        del new_options[CONF_CODE]
+        hass.config_entries.async_update_entry(entry, options=new_options)
```

Comment:
I discussed this with @frenck and we think we should have a proper [config entry migration](https://developers.home-assistant.io/docs/config_entries_config_flow_handler#config-entry-migration) here.

We don't support downgrades.
However, suppose a user downgrades with a proper config entry migration. In that case, HA will inform the user that the config entry is not compatible as the config entry version is not matching.
With the code above, the user will probably get some exceptions as you are deleting the `CONF_CODE` option.